lark/lark_validator.py

```diff
@@ -0,0 +1,265 @@
+from typing import Any, Dict, List
+
+from .exceptions import GrammarError
+from .grammar import TOKEN_DEFAULT_PRIORITY, RuleOptions
+from .lexer import Token
+from .load_grammar import eval_escaping
+from .tree import Tree
+
+class Definition:
+    def __init__(self, is_term, tree, params=(), options=None):
+        self.is_term = is_term
+        self.tree = tree
+        self.params = tuple(params)
+
+class LarkValidator:
```

Comment:
Please add a docstring here and in `validate`, something like:

```
Checks a grammar parsed by `lark.lark` for validated using a variety of checks similar to what `load_grammar.py` does on parser creation. The only stable public entry point is `LarkValidator.validate(tree)`

Checks:
- Illegal constructs not prevented by the grammar
  - `alias` not in the top expansions of a rule
  - Incorrect `%ignore` lines
  - Invalid literals (like newlines inside of regex without the `x` flag)
  - Rules used inside of Terminals
- Undefined symbols
- Incorrectly used templates
```

Reply:
Also we should make sure that this is exposed in the docs, @erezsh can probably say what needs to be done for that.

Reply:
Will do.

Reply:
Fixed by commit 654e102300faeeba277841c3fe0af75ad66f41e5.